src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/ExactTypeBindingProvider.cs

```diff
@@ -96,7 +102,14 @@ private class ExactBinding : BindingBase<TAttribute>
 
                 IValueProvider vp = new ConstantValueProvider(obj, typeof(TUserType), invokeString);
                 return vp;
-            }            
+            }
+
+            protected override Task<IValueProvider> BuildAsync(TAttribute attrResolved, BindingContext context)
+            {
+                _preBindHook?.Invoke(attrResolved, context);
```

Comment:
What if you had the BindingContext passed into the `BindToExactType` delegate? Would you need the post-resolve/pre-bind hooks?

I don't like these hooks either -- I've been asking Mike if we can reduce them to keep the concept count lower. I think 
with his newest changes we'll be closer to the 'custom type' model that you're proposing.